packages/directlinespeech/src/createAdapters.js

```diff
@@ -0,0 +1,161 @@
+import {
+  BotFrameworkConfig,
+  DialogServiceConnector,
+  PropertyId,
+  ServicePropertyChannel
+} from 'microsoft-cognitiveservices-speech-sdk';
+
+import createWebSpeechPonyfillFactory from './createWebSpeechPonyfillFactory';
+import DirectLineSpeech from './DirectLineSpeech';
+import patchDialogServiceConnectorInline from './patchDialogServiceConnectorInline';
+import resolveFunctionOrReturnValue from './resolveFunctionOrReturnValue';
+
+const TOKEN_RENEWAL_INTERVAL = 120000;
+
+export default async function create({
+  audioConfig,
+  audioContext,
+  enableTelemetry,
+  fetchCredentials,
+  speechRecognitionEndpointId,
+  speechRecognitionLanguage = (typeof window !== 'undefined' &&
+    typeof window.navigator !== 'undefined' &&
+    window.navigator.language) ||
+    'en-US',
+  speechSynthesisDeploymentId,
+  speechSynthesisOutputFormat,
+  textNormalization,
+  userID,
+  username
+}) {
+  if (!fetchCredentials) {
+    throw new Error('"fetchCredentials" must be specified.');
+  }
+
+  const { authorizationToken, region, subscriptionKey } = await resolveFunctionOrReturnValue(fetchCredentials);
+
+  if (
+    (!authorizationToken && !subscriptionKey) ||
+    (authorizationToken && subscriptionKey) ||
+    (authorizationToken && typeof authorizationToken !== 'string') ||
+    (subscriptionKey && typeof subscriptionKey !== 'string')
+  ) {
+    throw new Error(
+      '"fetchCredentials" must return either "authorizationToken" or "subscriptionKey" as a non-empty string only.'
+    );
+  }
+
+  if (typeof enableTelemetry !== 'undefined') {
+    console.warn(
+      'botframework-directlinespeech: Telemetry options is not supported yet. Please refer to Cognitive Services documentation for details.'
```

Comment:
```suggestion
      'botframework-directlinespeech: Telemetry options are not yet supported. Please refer to Cognitive Services documentation for details.'
```

Do you want to add a link to this one?